akka-http-backend/src/main/scala/com/softwaremill/sttp/akkahttp/AkkaHttpBackend.scala

```diff
@@ -358,14 +385,15 @@ object AkkaHttpBackend {
       customHttpsContext: Option[HttpsConnectionContext] = None,
       customConnectionPoolSettings: Option[ConnectionPoolSettings] = None,
       customLog: Option[LoggingAdapter] = None
+  )(
+      http: AkkaHttpClient = AkkaHttpClient.fromAkkaHttpExt(actorSystem, customHttpsContext, customLog)
   )(implicit ec: ExecutionContext = ExecutionContext.Implicits.global): SttpBackend[Future, Source[ByteString, Any]] =
-    AkkaHttpBackend(
+    make(
       actorSystem,
       ec,
       terminateActorSystemOnClose = false,
```

Comment:
LMK what you think about solving this by having a new backend class that wraps over an existing backend but also executes some shutdown hooks. This will allow us to completely hide ActorSystem from AkkaHttpBackend (only Materializer and Config will be needed)

Reply:
Not sure if I see the benefit - apart from one field less in the backend? Would that "raw" backend be somehow reusable in other contexts?

Reply:
Probably not ;) just using a less powerful abstraction.

Reply:
OK, now I have a benefit - no need to start an actor system at all to create the testing backend...